src/matrix.h

```diff
@@ -15,107 +15,103 @@ namespace bla
         ColMajor,
         RowMajor
     };
-    template <typename T, ORDERING ORD = ORDERING::ColMajor>
-    class Matrix
+    template <typename T, ORDERING ORD>
+    class MatrixView : public MatExpr<MatrixView<T, ORD>>
     {
-        size_t rows_, cols_;
-        std::shared_ptr<T[]> data_;
-        bool isTransposed_ = false;
+    protected:
+        size_t rows_, cols_, dist_;
```

Comment:
dist_ should be a integral constant 1 if unused? (see vector.h)

Reply:
no, dist_ is either the number of rows or cols depending on the ordering.